src/PropertyHandler.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Livewire;
+
+abstract class PropertyHandler
+{
+    /**
+     *  When an update is triggered, a request is sent to the server INCLUDING the last-known component state.
+     *  The server "hydrates" or "deserializes" the component from that state and performs any updates.
+     */
+    abstract public static function hydrate($value);
+
+    /**
+     * In addition to render HTML, Livewire "dehydrates" or "serializes" the component's state
+     * of a public property, so it can be passed to the front-end.
+     */
+    abstract public function dehydrate($value);
```

Comment:
Just had a thought.. maybe these methods should also be passed the `ReflectionProperty` instance? This way the consumer can leverage PHP 8 attributes in their handler. Being able to do something like this would be awesome:
```php
#[DateFormat('H:i, d M Y')]
public ?Carbon $date = null;
```

and then the handler can check this:
```php
public function dehydrate($value, ReflectionProperty $property)
{
    $property->getAttributes();
    // ...
}
```

Reply:
@jameslkingsley yeah I like it. I think in the interest in keeping the PR focused @jonassiewertsen, maybe we will hold off on this and do it in a seperate PR later once this is merged, where we can flesh out a detailed description of how it can be useful and how it will work with clear examples.

Reply:
I changed this idea in my PR to pass the component instance and property name, eg:
```php
public function hydrate($instance, $key, $value)
```
This gives the option to build the reflection instance manually or anything else the developer might find useful.

Reply:
+1 for component instance and property name, even better would be to use the same signatures that hydrateModel, hydrateModels, dehydrateModel and dehydrateModels currently use:
```php
function hydrate($value, $property, $request, $instance);
function dehydrate($value, $property, $response, $instance);
```
I have a piece of work in progress that is currently heavily hacking HydratePublicProperties.php, it would be nice to re-implement it with this, but it needs the instance and property name, along with the request/response (for hydration and dehydration respectively).